homeassistant/components/google_assistant/trait.py

```diff
@@ -1316,18 +1316,33 @@ def query_attributes(self):
     async def _execute_volume_relative(self, data, params):
         # This could also support up/down commands using relativeSteps
         relative = params["volumeRelativeLevel"]
-        current = self.state.attributes.get(media_player.ATTR_MEDIA_VOLUME_LEVEL)
 
-        await self.hass.services.async_call(
-            media_player.DOMAIN,
-            media_player.SERVICE_VOLUME_SET,
-            {
-                ATTR_ENTITY_ID: self.state.entity_id,
-                media_player.ATTR_MEDIA_VOLUME_LEVEL: current + relative / 100,
-            },
-            blocking=True,
-            context=data.context,
-        )
+        # if we have access to current volume level, do a single 'set' call
+        if media_player.ATTR_MEDIA_VOLUME_LEVEL in self.state.attributes:
```

Comment:
This should not have been merged in it's current state. It doesn't work.

If you see the `supported` method of this trait, it is only added with entities that have `SUPPORT_VOLUME_SET`, which is the exact method that was used here. Because of that, it will never happen that an entity receives an execute that we cannot set volume off. If we are, it's a bug in the integration that provides the entity.

This would have been caught in the test, except that the test is based off the other media player test, **minus** the check for `assert trait.VolumeTrait.supported(…)`. 

Volume up and down are available if the entity has `SUPPORT_VOLUME_STEP`, so that needs to be added to `supported`, used in the if-statement in the execute and tested for in the test.